src/lib/workers/worker.js

```diff
@@ -33,184 +36,171 @@ function expireCache() {
+                            let handlebarInstance = GetHandlebarsInstance(dataTypeHandler, templatesMap);
+                            session.set(constants.CLS_KEY_HANDLEBAR_INSTANCE, handlebarInstance);
+                            session.set(constants.CLS_KEY_TEMPLATE_LOCATION, path.join(constants.TEMPLATE_FILES_LOCATION, dataTypeHandler.dataType));
+
+                            dataTypeHandler.parseSrcData(s)
+                                .then((parsedData) => {
+                                    var dataContext = { msg: parsedData };
+                                    if (templateString == null || templateString.length == 0) {
+                                        var result = Object.assign(dataTypeHandler.getConversionResultMetadata(dataContext.msg),
+                                            JSON.parse(JSON.stringify(dataContext.msg)));
+
+                                        fulfill({ 'status': 200, 'resultMsg': result });
+                                    }
+                                    else {
+                                        var template = handlebarInstance.compile(dataTypeHandler.preProcessTemplate(templateString));
 
-                    if (!messageContent || messageContent.length == 0) {
-                        reject({ 'status': 400, 'resultMsg': errorMessage(errorCodes.BadRequest, "No message provided.") });
-                    }
+                                        try {
+                                            fulfill({ 'status': 200, 'resultMsg': generateResult(dataTypeHandler, dataContext, template) });
+                                        }
+                                        catch (err) {
+                                            reject({ 'status': 400, 'resultMsg': errorMessage(errorCodes.BadRequest, "Unable to create result: " + err.toString()) });
+                                        }
+                                    }
 
-                    var msgObject = {};
-                    try {
-                        msgObject = hl7.parseHL7v2(messageContent);
-                    }
-                    catch (err) {
-                        reject({
-                            'status': 400,
-                            'resultMsg': errorMessage(errorCodes.BadRequest, "Unable to decode and parse HL7 v2 message. " + err.toString())
-                        });
+                                })
+                                .catch(err => {
+                                    reject({ 'status': 400, 'resultMsg': errorMessage(errorCodes.BadRequest, `Unable to parse input data. ${err.error.message}`) });
+                                });
+                        }
+                        catch (err) {
+                            reject({ 'status': 400, 'resultMsg': errorMessage(errorCodes.BadRequest, `${err.toString()}`) });
+                        }
                     }
+                    break;
+
+                case '/api/convert/:template':
```

Comment:
This is different than what is in the routes file. There is a srcDataType in Routes.js line 773, but not one here.

Reply:
it's just a type to pass message (and it's same as documented path).